mp4parse/src/boxes.rs

```diff
@@ -70,23 +70,27 @@ impl From<BoxType> for FourCC {
     }
 }
 
-impl<'a> From<&'a str> for FourCC {
-    fn from(v: &'a str) -> FourCC {
-        FourCC {
-            value: v.to_owned(),
-        }
+impl From<[u8; 4]> for FourCC {
+    fn from(v: [u8; 4]) -> FourCC {
+        FourCC { value: v }
     }
 }
 
 impl fmt::Debug for FourCC {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.value)
+        match std::str::from_utf8(&self.value) {
+            Ok(s) => write!(f, "{}", s),
+            Err(_) => self.value.fmt(f),
```

Comment:
This is a slight change, but I figure it's more useful to see the non-utf8 values in debugging rather than "null".